hazelcast/src/main/java/com/hazelcast/instance/impl/executejar/ExecuteJobParameters.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (c) 2008-2023, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.instance.impl.executejar;
+
+import com.hazelcast.jet.Job;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.ArrayList;
+
+public class ExecuteJobParameters {
+
+    @Nonnull
+    private final String jarPath;
+    @Nullable
+    private final String snapshotName;
+    @Nullable
+    private final String jobName;
+
+    private final ArrayList<Job> submittedJobs = new ArrayList<>();
```

Comment:
not sure if "job parameters" object is the best place to hold submitted jobs, but I have no idea for now for. better place

Reply:
All the state must go into threadlocal object. That's the purpose